### DISCUSSION TITLE
Issue with 1.23 docker image

### DISCUSSION THREAD

**Discussion by woolmonkey** [dani-garcia/vaultwarden#2054] [Q&A]
Upgraded to 1.23 in the docker image running mysql backend and get this error in the logs
"Can't create table `bitwarden`.`emergency_access` (errno: 150 \"Foreign key constraint is incorrectly formed\")",

When I role back to 1.22.2 the error is not there.

**Comment by BlackDex** [comment]
Your database charset and/or collation isn't correct.
See this post on how to resolve that https://github.com/dani-garcia/vaultwarden/discussions/1559#discussioncomment-544070

**Reply by FuzzyMistborn** [reply]
You know, I was the one who originally raised that issue/discussion and I implemented that fix.  But I'm running into this issue again.  Any way I could make this "permanent" so I don't have to keep doing it every time there's a new database row?

Edit: to clarify, I had done the steps in #1559 previously.  I got the same error when updating to 1.23.  I re-ran the commands (though I got nothing on output from the second set "SELECT_CONCAT" so stopped there) and it worked.

**Reply by BlackDex** [reply]
I think you probably didn't update the defaults then, if you forgot run
```mysql
ALTER DATABASE `bitwarden_rs` CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
```
then new tables will still have the wrong charset and collate, because they use the default setting from the database.

**Reply by FuzzyMistborn** [reply]
No, I ran all the commands.  Guess we'll see what happens next time.

**Reply by BlackDex** [reply]
I just tested what happens if i have a MySQL/MariaDB server which defaults to `latin1` and i created a new database which tells it to be `utf8mb4`, all tables are created without issues and all having the correct charset and collation.

If i undo the latest migration, in this case the emergency access, and configure the database to be `latin1` again, the new migration fails.
If i then reconfigure the database back to `utf8mb4` and start Vaultwarden again, it runs just fine.

This is not something Vaultwarden changes at runtime. It looks like something happened with your database in between the previous time you run this and the latest attempt. Maybe you restored a backup, if using docker for mysql, maybe that did something. I don't know.
Lets just hope it will work without issues the next time there are migrations needed.

**Reply by FuzzyMistborn** [reply]
Agreed.  I may have done a restore, can't recall.  But it would have been a recent backup (after the last migration).  Who knows.  At least it works as you said for now.
